pkg/ans/ans_test.go

```diff
@@ -0,0 +1,223 @@
+package ans
+
+import (
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/xsuaa"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestANS_Send(t *testing.T) {
+	testXSUAA := xsuaa.XSUAA{
+		OAuthURL:     "https://my.test.oauth.provider",
+		ClientID:     "myTestClientID",
+		ClientSecret: "super secret",
+		CachedAuthToken: xsuaa.AuthToken{
+			TokenType:   "bearer",
+			AccessToken: "1234",
+			ExpiresIn:   12345,
+		},
+	}
+	type request struct {
+		path       string
+		authHeader string
+		event      Event
+	}
+	tests := []struct {
+		name        string
+		event       Event
+		wantErrf    string
+		wantRequest request
+	}{
+		{
+			name: "Straight forward",
+			event: Event{
+				EventType:      "my event",
+				EventTimestamp: 1647526655,
+			},
+			wantRequest: request{
+				path:       "/cf/producer/v1/resource-events",
+				authHeader: "bearer 1234",
+				event: Event{
+					EventType:      "my event",
+					EventTimestamp: 1647526655,
+				},
+			},
+		},
+		{
+			name:     "Wrong status code in response error",
+			wantErrf: "http request to '%s/cf/producer/v1/resource-events' did not return expected status code 202; instead got 200",
```

Comment:
So when you don't give the ANS send function an event to send, it still succeeds because it's able to access the url? Should we throw an error here instead? Or is there a case where we want it to succeed even if is called without an event?

Reply:
The http.Do function succeeds even without event. Then the status code returned is checked, which will be some error code, as the ANS server replies that there was no event json. The check for status code to be 202 **is the error check**. Does that make sense?

Reply:
Ah okay, but it's only the error check for the test, right? The user would just see the logs that there was a 200 status code and they wouldn't have received a message, but there's no error sent to the user saying something like "No event given" or something like that?

Reply:
That is true. Only the status code from the server to the user. To find the reason the body needs to be read and returned. If you want I can build that in?

Reply:
That might be nice for debugging? But what do you think? Would there be a use case where someone could even influence something to make it so that an event is sent when it wasn't before? If not, and it's only a coding problem that would prevent events from being sent then I think we're okay without the extra information.

Reply:
I added the [reading of the response body in case of an unforseen response code](https://github.com/SAP/jenkins-library/pull/3654/files#diff-61d78fe7e36da0bdc23c544ab2c50d0a411020424a26e4529cd53013d9d42eabR120)

Reply:
Looks great, thanks!